### ISSUE TITLE
(@aws-cdk/aws-codepipeline-actions): No AppConfigDeployAction available 

### ISSUE THREAD

**Issue by JunChatani** [aws/aws-cdk#23598]
### Describe the feature

For CodePipeline there is no AppConfig deployment action available.

### Use Case

We have pipelines that are solely deploying appConfig, json parameters.
(SSM parameters are not yet supported, we validated this by contacting internal teams through a support case).

### Proposed Solution

Simplified construct, perhaps prefixing it with Cfn like all other AppConfig resources. 
The pipeline role will also need a policy allowing appConfig actions.
```
import * as codepipeline from "@aws-cdk/aws-codepipeline";
import { Construct } from "constructs";
import { Action } from "../action";
import { deployArtifactBounds } from "../common";

export interface AppConfigDeployActionProps extends codepipeline.CommonAwsActionProps {
    readonly actionName: string;
    readonly inputs: codepipeline.Artifact[]

    readonly applicationId: string
    readonly environmentId: string
    readonly configurationProfileId: string
    readonly deploymentStrategyId: string
    readonly inputArtifactConfigurationPath: string
}

class AppConfigDeployAction extends Action {
    protected readonly providedActionProperties: AppConfigDeployActionProps

    constructor(providedActionProperties: AppConfigDeployActionProps) {
        super({
            ...providedActionProperties,
            category: codepipeline.ActionCategory.DEPLOY,
            provider: 'AppConfig',
            artifactBounds: deployArtifactBounds(),
        });

        this.providedActionProperties = providedActionProperties;
    }

    protected bound(scope: Construct, _stage: codepipeline.IStage, _options: codepipeline.ActionBindOptions): codepipeline.ActionConfig {
        return {
            configuration: {
                Application: this.providedActionProperties.applicationId,
                Environment: this.providedActionProperties.environmentId,
                ConfigurationProfile: this.providedActionProperties.configurationProfileId,
                DeploymentStrategy: this.providedActionProperties.deploymentStrategyId,
                InputArtifactConfigurationPath: this.providedActionProperties.inputArtifactConfigurationPath,
            },
        };
    }

}
```

### Other Information

We wanted to port [this example piece of Kotlin](https://github.com/aws-samples/aws-appconfig-codepipeline-cdk/blob/main/infrastructure/src/main/kotlin/com/app/config/AppConfigCiCdStack.kt) code to ts, but it was not a 1-on-1 mapping. 

AppConfig is still prefixed with Cfn, thus no priority is given towards this service. However it is rather a small request and may be useful for others as well. 

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.59.0

### Environment details (OS name and version, etc.)

/

**Comment by peterwoodworth** [comment]
It would be great to have a class which does this @JunChatani! Thanks for the request 🙂 

I am marking this issue as p2, which means that we are unable to work on this immediately.

We use +1s to help prioritize our work, and are happy to revaluate this issue based on community feedback. You can reach out to the cdk.dev community on [Slack](https://cdk.dev) to solicit support for reprioritization.

**Comment by JunChatani** [comment]
Hi @peterwoodworth thanks for the reply, is it allowed for beginning contributors to submit a PR for medium effort requests?

**Comment by peterwoodworth** [comment]
Yes of course! Feel free to get started, check out our [contributing guide](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md) for more info, and be sure to check how existing codepipeline actions are designed 🙂

**Comment by JunChatani** [comment]
I tried an attempt by more or less copying the Beanstalk Deploy action PR, but hit a wall regarding the integration test of it. 

AppConfig does [not yet have L2 constructs](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_appconfig-readme.html), making it difficult to write a clean integration test without declaring any AppConfig Class that extends a Construct. This should probably be postponed until there are official AppConfig L2 constructs.
